apps/api/src/controllers/post.ts

```diff
@@ -0,0 +1,128 @@
+import Post from '../models/post';
+
+// Initialize posts array to save data in memory
+const posts = [];
+
+// Get all posts
+export const getPosts = async (req, res) => {
+  try {
+    // Return all the posts with a 200 status code
+    const posts = await Post.find();
+
+    res.status(200).json(posts);
+  } catch (error) {
+    const { message } = error;
+    res.status(500).json({ message });
+  }
+};
+
+const getPost = async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const post = await Post.findById(id);
+
+    if (!post) {
+      return res.status(404).json({ message: 'Post not found!' });
+    }
+    res.status(200).json(post);
+  } catch (error) {
+    const { message } = error;
+    res.status(500).json({ message });
+  }
+};
+
+// Get post by category
+const getPostsByCategory = async (req, res) => {
+  // Retrieve the id from the route params
+  const { category, id } = req.params;
+  try {
+    const postFound = await Post.findOne(category);
+
+    if (!postFound) {
+      // If we don't find the post return a 404 status code with a message
+      return res.status(404).json({ message: 'Post by Category not found' });
+      // Note: Remember that json method doesn't interrupt the workflow
+      // therefore is important to add a "return" to break the process
+    }
+
+    // Return the post with a 200 status code
+    res.status(200).json(postFound);
+  } catch (error) {
+    const { message } = error;
+    res.status(500).json({ message });
+  }
+};
+
+// Create post
+const createPost = async (req, res) => {
+  try {
+    const post = await Post.create(req.body);
+    // Return the created post with a 201 status code
+    res.status(201).json(post);
+  } catch (error) {
+    const { message } = error;
+    res.status(500).json({ message });
+  }
+};
+
+// Create comment
+const createComment = async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const post = await Post.findByIdAndUpdate(id, req.body, { new: true });
```

Comment:
This method is the same as the `updatePost` what you need to do here is get the post, then create a new Comment with the schema `Comment.create(...)`, then obtain the ID of the comment and you can push it to the post instance and save it `post.comments.push(commentId)` and `await post.save()`, with this you will save the ID of the comment as a reference in you post document